src/greynirseq/utils/preprocessing/filters.py

```diff
@@ -0,0 +1,997 @@
+            if len(w) > cls.auto_pass_len and not cls.num_prog.match(w)
+        ]
+        eng_words = [
+            w.lower()
+            for w in eng_words
+            if len(w) > cls.auto_pass_len and not cls.num_prog.match(w)
+        ]
+        cls._store["ice"].update(ice_words)
+        cls._store["eng"].update(eng_words)
+
+        if cls._prev_store:
+            pice = all(
+                cls._prev_store["ice"].get(w, 0) >= cls.min_freq for w in ice_words
+            )
+            peng = all(
+                cls._prev_store["eng"].get(w, 0) >= cls.min_freq for w in eng_words
+            )
+            return pice and peng
+        return True
+
+
+
+@register_filter
+def wrong_quotes(ex):
+    ice, eng = ex["is"], ex["en"]
+
+    QUOTES = set(QUOTE_LIKE)
+    QUOTES.remove("'")
+    QUOTES.remove('"')
+    QUOTES.remove(",")
+    for q in ICE_QUOTE.ALL:
+        try:
+            QUOTES.remove(q)
+        except:
+            pass
+
+    chars_ex = set(ice)
+    chars_ex.update(eng)
+
+    return not bool(chars_ex & QUOTES)
+
+
+# @register_filter
+# def language(ex):
+#     ice, eng = ex["is"], ex["en"]
+#     correct = probably_correct_language(ice, "is", lower_bound=0.995)
+#     correct = correct and probably_correct_language(eng, "en", lower_bound=0.995)
+#     return correct
+
+
+class Pipeline:
+
+    counter = dict()
+    _fns = [
+        ### filters
+        null_sentence,
+        alphanumeric,
+        banned_symbol,
+        whitelist_symbol,
+        digit_mismatch,
+        deduplicate,
+        #case_mismatch,
+        max_word_length,
+        min_word_count,
+        ### transformations
+        fix_improper_line_split,
+        #remove_leading_bullet,
+        soft_hyphen,
+        replace_dashes,
+        merge_spaces,
+        # fix_ice_quotes,
+        ### filters
+        # bullet_mismatch,
+        #quote_mismatch,
+        ocr_wrong_symbol,
+        # colon_mismatch,
+        #missing_letter,
+        #corrupt_symbol,
+        quote_inside_word,
+        abs_min_string_edit,
+        rel_min_string_edit,
+        #abs_min_subtoken_edit,
+        #rel_min_subtoken_edit,
+        sentence_length_ratio,
+        strict_sentence_length_ratio,
+        #token_count_ratio,
+        ocr_word_boundary_avg_length,
+        #dot_pattern,
+        # wrong_quotes,
+        ###############
+        # language,
+        # MinFrequency.gather,
+        # MinFrequency,
+        # MostCommon50k,
+    ]
+    start_time = None
+    end_time = None
+
+    @classmethod
+    def run(cls, examples, view_function=None, inverted=False, **kwargs):
+        cls.start_time = time.time()
+        cls.counter.clear()
+        cls.counter["total"] = 0
+        for ex in examples:
+            cls.counter["total"] += 1
+            ex = cls.process(ex, view_function=view_function, inverted=inverted)
+            if ex is not None:
+                yield ex
+
+        for obj in cls._fns:
+            if not isinstance(obj, type):
```

Comment:
A bit funny check here. Is there a base class having the save_to_file() member? If so, this should probably check for that base class.

Reply:
Yeah, it's supposed to check for the `
Gather` class.